crates/uv-installer/src/plan.rs

```diff
@@ -272,6 +262,36 @@ impl<'a> Planner<'a> {
                                 }
                             }
                         }
+                        Dist::Source(SourceDist::DirectUrl(sdist)) => {
+                            // Find the most-compatible wheel from the cache, since we don't know
+                            // the filename in advance.
+                            if let Some(wheel) = built_index.url(&sdist)? {
+                                let cached_dist = wheel.into_url_dist(url.clone());
+                                debug!("URL source requirement already cached: {cached_dist}");
+                                cached.push(CachedDist::Url(cached_dist));
+                                continue;
+                            }
+                        }
+                        _ => unreachable!(),
```

Comment:
These `Unreachable` seem like a sign that something is off in the data model. How could we refactor to avoid them?